nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/RunMongoAggregation.java

```diff
@@ -121,6 +126,7 @@
 
         final Set<Relationship> _relationships = new HashSet<>();
         _relationships.add(REL_RESULTS);
+        _relationships.add(REL_EMPTY);
```

Comment:
Could you instead add a property like GetMongo has "Send Empty Result"?

Reply:
In the GetMongo processor there's an option to "Send Empty Result" instead of creating another relationship, what about adding this option instead for symmetry. This option sends an empty JSON object out the success relationship if there's no result.

I had the same issue with GetMongo and the "Send Empty Result" fixed my issue.

Reply:
Ok. I've changed to write an empty flowfile instead a new relationship.